src/cursor/cursor.ts

```diff
@@ -55,6 +57,83 @@ export interface CursorOptions extends CoreCursorOptions {
   numberOfRetries?: number;
 }
 
+export class CursorStream extends Readable {
+  [kCursor]: Cursor;
+
+  /** @event */
+  static readonly CLOSE = 'close' as const;
+  /** @event */
+  static readonly DATA = 'data' as const;
+  /** @event */
+  static readonly END = 'end' as const;
+  /** @event */
+  static readonly FINISH = 'finish' as const;
+  /** @event */
+  static readonly ERROR = 'error' as const;
+  /** @event */
+  static readonly PAUSE = 'pause' as const;
+  /** @event */
+  static readonly READABLE = 'readable' as const;
+  /** @event */
+  static readonly RESUME = 'resume' as const;
+
+  constructor(cursor: Cursor) {
+    super({ objectMode: true });
+    this[kCursor] = cursor;
+    this.on('end', () => this[kCursor].close(() => this.emit(Cursor.CLOSE)));
+  }
+
+  destroy(err?: AnyError): void {
+    if (err) this.emit(Cursor.ERROR, err);
+    this.pause();
+    this[kCursor].close(() => this.emit(Cursor.CLOSE));
+  }
+
+  /** @internal */
+  _read(): void {
+    const cursor = this[kCursor];
+    if ((cursor.s && cursor.s.state === CursorState.CLOSED) || cursor.isDead()) {
+      this.push(null);
+      return;
+    }
+
+    // Get the next item
+    cursor._next((err, result) => {
+      if (err) {
+        if (this.listeners(CursorStream.ERROR) && this.listeners(CursorStream.ERROR).length > 0) {
+          this.emit(CursorStream.ERROR, err);
+        }
+        if (!cursor.isDead()) cursor.close();
```

Comment:
should we be waiting for `cursor.close` to complete before emitting these events?